test_suite/test_filenames.py

```diff
@@ -0,0 +1,19 @@
+import os
+import pytest
+import shutil
+
+@pytest.fixture(scope='session')
+def testing_directory(tmpdir_factory):
+    root_a = tmpdir_factory.mktemp('A')
+    root_a.join('wow.txt').write('h')
+    root_a.join('A1.csv').write('h')
+
+    root_b = tmpdir_factory.mktemp('B')
+    root_b.join('hey.txt').write('h')
+    return tmpdir_factory
```

Comment:
This should return the path to a root folder inside the temporary directory and not `tmpdir_factory`. This is because `tmpdir_factory` is a session-scoped fixture, so it doesn't necessarily behave the way you would expect between tests. 

For an example of what I mean, try running the tests below with `python -m pytest -s`:

```
import pytest
import os

@pytest.fixture(scope='session')
def testing_directory1(tmpdir_factory):
    root_a = tmpdir_factory.mktemp('A')
    root_a.join('wow.txt').write('h')
    root_a.join('A1.csv').write('h')
    return tmpdir_factory

def test_dir1(testing_directory1):
    contents = os.listdir(testing_directory1.getbasetemp().strpath)
    print(f"The contents of tmpdir_factory.getbasetemp().strpath called from inside the test:\n{contents}")

@pytest.fixture(scope='session')
def testing_directory2(tmpdir_factory):
    root_a = tmpdir_factory.mktemp('A')
    root_a.join('wow.txt').write('h')
    root_a.join('A1.csv').write('h')
    return root_a

def test_dir2(testing_directory2):
    contents = os.listdir(testing_directory2)
    print(f"The contents of root_a called from inside the test:\n{contents}")
```

This gives me the output:

```
platform linux -- Python 3.8.0, pytest-6.0.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/chris/projects/fca
collected 6 items                                                                                                                                                                                                                                                                                             

test/test_fca_algorithms.py ....The contents of tmpdir_factory.getbasetemp().strpath called from inside the test:
['Acurrent', 'A0', '.lock']
.The contents of root_a called from inside the test:
['A1.csv', 'wow.txt']
.
```

As you can see, the contents of `testing_directory.getbasetemp().strpath` is not `[A. B]` as we would probably expect, and in fact neither of those folders exists.

If we make this fixture return the path to a root temp directory that we create using `tmpdir_factory` then it should work as intended. This also has the benefit of allowing us to make this a class or function scoped fixture which means we are less likely to encounter files from other tests in our temporary dirs as cleanup happens after each test (or set as grouped by class).

Reply:
It might make it clearer to look at [the API for tmp_path_factory](https://docs.pytest.org/en/stable/reference.html#pytest.tmpdir.TempPathFactory.mktemp) which is practically identical to `tmpdir_factory` (in fact maybe we should switch? idk). Specifically the description of `tmp_path_factory.mktemp` 

> Create a new temporary directory managed by the factory. 

The `tmp_path_factory` is a factory class for constructing temporary directories/files and manages them afterwards, so it's not really meant to be used like a directory itself. Under the hood, the temporary directories are numbered for uniqueness, so if you run two tests with the fixture `testing_directory` you have actually made two folders `A0` and `A1`.